pybdv/bdv_datasets.py

```diff
@@ -0,0 +1,213 @@
+
+import os
+import numpy as np
+import json
+from .util import open_file
+from .converter import make_bdv
+from shutil import rmtree
+
+
+def _check_for_out_of_bounds(position, volume, full_shape, verbose=False):
+
+    position = np.array(position)
+    full_shape = np.array(full_shape)
+
+    vol_shape = np.array(volume.shape)
+    if position.min() < 0 or (position + vol_shape - full_shape).max() > 0:
+
+        print(f'position = {position}')
+
+        too_large = (position + vol_shape - full_shape) > 0
+        source_ends = vol_shape
+        source_ends[too_large] = (full_shape - position)[too_large]
+
+        source_starts = np.zeros((3,), dtype=int)
+        source_starts[position < 0] = -position[position < 0]
+        position[position < 0] = 0
+
+        if verbose:
+            print(f'source_starts = {source_starts}')
+            print(f'source_ends = {source_ends}')
+            print(f'position = {position}')
+
+        volume = volume[
+            source_starts[0]: source_ends[0],
+            source_starts[1]: source_ends[1],
+            source_starts[2]: source_ends[2]
+        ]
+
+    return position, volume
+
+
+def _check_shape_and_position_scaling(max_scale, position, volume,
+                                      data_path, setup_id, timepoint,
+                                      verbose=False):
```

Comment:
You can just pass the key from https://github.com/constantinpape/pybdv/blob/master/pybdv/util.py#L38 to this function to make it work for h5 and n5 and make it more readable.

Reply:
Did exactly that